plugin.video.sms/service.py

```diff
@@ -0,0 +1,175 @@
+"""
+    Scoot Media Streamer
+    Copyright (C) 2020  Scott Ware
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+################################################################################################
+
+from operator import itemgetter
+from bottle import route, run
+from threading import Thread
+import os
+import sys
+import copy
+import urllib
+import requests
+import time
+import uuid
+import json
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xbmc
+
+################################################################################################
+
+addon = xbmcaddon.Addon('plugin.video.sms')
+addon_path = addon.getAddonInfo('path')
+libs = xbmc.translatePath(os.path.join(addon_path, 'resources', 'lib'))
+sys.path.append(libs)
+
+################################################################################################
+
+import sms
+import client
+import bottle_ext
+
+################################################################################################
+
+CLIENT = 4
+FORMATS = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18]
+CODECS = [10,11,12,13,20,30,31,32,40,1000,1001,1002,1003,1004,1005,1006,1007,1008,1009,1010,1011,2000,2001,2002,2003,2004]
+MCH_CODECS = [1000,1001,1002,1003,1004,1005,1006,1008,1009,1010,1011]
+FORMAT = 9
+
+class Service(object):
+    def __init__(self):
+        # Settings
+        self.settings = {\
+            'serverUrl': addon.getSetting('serverUrl'), \
+            'serverPort': addon.getSetting('serverPort'), \
```

Comment:
you might want to explore our new functions to get typed functions: getSettingBool, getSettingInt, etc. See: https://codedocs.xyz/xbmc/xbmc/group__python__xbmcaddon.html